site/src/sphinx/server-annotated-service.rst

```diff
@@ -748,6 +749,38 @@ order commented. It is also the same as the evaluation order of the converters.
                         new MyGlobalExceptionHandler5(),        // order 5
                         new MyGlobalExceptionHandler6());       // order 6
 
+Returning a response
+--------------------
+
+In the earlier examples, the annotated service methods only return :api:`HttpResponse`, however there are
+more response types which can be used in the annotated service.
+
+- :api:`HttpResponse` and :api:`AggregatedHttpMessage`
+
+  - It will be sent to the client without any modification. If an exception is raised while the response is
+    being sent, exception handlers will handle it. If no message has been sent to the client yet,
+    the exception handler can send an :api:`HttpResponse` instead.
+
+- ``Publisher<?>``
+
+  - All objects which are produced by the publisher will be collected, then the collected ones will be
+    converted to an :api:`HttpResponse` by response converters. If a single object is produced, it will be
+    passed into the response converters as it is. But if multiple objects are produced, they will be passed
+    into the response converters as a list. If the producer produces an error, exception handlers will handle it.
+    Note that ``ObservableSource<?>`` will be treated in the same way as ``Publisher<?>`` if you use
+    ``armeria-rxjava``.
+
+- ``CompletionStage<?>``
```

Comment:
`CompletionStage` and `CompletableFuture`

(Just in case people don't know `CompletableFuture` is a subtype of `CompletionStage` :-))